### ISSUE TITLE
Relative-time view functions don't update

### ISSUE THREAD

**Issue by garrett-gottlieb** [mobxjs/mobx-state-tree#1520]
**_Bug report_**

* [X] I've checked documentation and searched for existing issues
* [X] I've made sure my project is based on the latest MST version
* [ ] Fork [this](https://codesandbox.io/s/y64pzxj01) code sandbox or another minimal reproduction. 

**Sandbox link or minimal reproduction code**

```js
types.model({ myDate: types.Date }).views(self => ({ relativeDate() { return Date.now() - self.myDate.getTime() } }))
```

**Describe the expected behavior**

`relativeDate` should update or show some sort of error while accessing or with TypeScript.

**Describe the observed behavior**

`relativeDate` is frozen to the value when it's first called, as long as `myDate` doesn't change.

**Comment by mweststrate** [comment]
Please provide a reproduction as requested in the issue template

**Comment by garrett-gottlieb** [comment]
https://codesandbox.io/s/mobx-state-tree-todolist-ywei0?file=/index.js

**Comment by mweststrate** [comment]
1. Just doing a single autorun will never update, at least you should wrap
it in autorun or something similar to make it re-executable
2. views will cache and only update if their backing data changes. However,
Date.now() isn't something observable, so it can't be watched by MobX. To
get an observable timestamp, check https://github.com/mobxjs/mobx-utils#now

**Comment by garrett-gottlieb** [comment]
This is amazing. So by using `mobxUtils.now()` in a view, it will automatically cause the component using that property to render? I'm not sure about why `autorun` is necessary here, or how to use it inside of a view. Currently, I was using `setTimeout` in the component but it looks like that should be unnecessary as well.

**Comment by garrett-gottlieb** [comment]
After a bunch of testing, I found out that not only did I need to use `mobxUtils.now(interval)`, but I also had to wrap subcomponents in `observer` in order for them to update according to the interval. I didn't have to use `autorun`, although I think you were referring to using that in my sandbox example. It looks like MST uses `autorun` internally, and as such, I've never used it before myself.
